resolwe/flow/managers/local.py

```diff
@@ -7,12 +7,18 @@
 """
 from __future__ import absolute_import, division, print_function, unicode_literals
 
+import subprocess
+
 from .base import BaseManager
 
 
 class Manager(BaseManager):
     """Local manager for job execution."""
 
-    def run(self, data_id, script, priority='normal', run_sync=False, verbosity=1):
+    def run(self, data_id, dest_dir, argv, priority='normal', run_sync=False, verbosity=1):
         """Run process locally."""
-        self.executor.run(data_id, script, verbosity=verbosity)
+        subprocess.Popen(
+            argv,
```

Comment:
Hm, so this just allows executing of arbitrary processes? Can we move argv construction to this (per-manager) run method instead? I am not very comfortable with having functions which allow arbitrary shell execution.

Reply:
Not entirely arbitrary. The argument vector is composed by the manager itself, in the base class. The execution queue is a local variable in `BaseManager._communicate()`. The `run()` method itself is technically public but only used by the manager, as far as I know. I decided to generate the vector in the base class, since that has the most environmental information and lots of logic (which has to be common to all managers) needed to construct it, such as the executor name, target directory path and knowledge of the exact format of the executor command. If anything, I think the method should be made private, the manager's "frontend" is `communicate()` anyway.